builder/driver.go

```diff
@@ -0,0 +1,221 @@
+package builder
+
+import (
+	"context"
+
+	"github.com/docker/buildx/driver"
+	ctxkube "github.com/docker/buildx/driver/kubernetes/context"
+	"github.com/docker/buildx/store"
+	"github.com/docker/buildx/store/storeutil"
+	"github.com/docker/buildx/util/dockerutil"
+	"github.com/docker/buildx/util/imagetools"
+	"github.com/docker/buildx/util/platformutil"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/util/grpcerrors"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+)
+
+type Driver struct {
+	Name        string
+	Driver      driver.Driver
```

Comment:
:thinking: this is confusing :cry: 

I think we should avoid having `driver.Driver` and `builder.Driver` - I guess that was the reason for `driverInfo` before. I think it's ok for now, but maybe we should revisit later.

Reply:
Yes, ideally it should be directly carried by the factory.

Reply:
Yeah sounds better, changes to `Node`.